test_elasticsearch/test_connection.py

```diff
@@ -70,6 +70,62 @@ def test_ssl_context_and_depreicated_values(self):
         self.assertRaises(ImproperlyConfigured, Urllib3HttpConnection, ssl_context=ctx, ca_certs="/some/path/to/cert.crt")
         self.assertRaises(ImproperlyConfigured, Urllib3HttpConnection, ssl_context=ctx, ssl_version=ssl.PROTOCOL_SSLv23)
 
+class TestGzipEnabledConnection(TestCase):
```

Comment:
these tests are only testing logic from the `Urllib3HttpConnection`, which is already tested and there is no need to retest it. What we need instead are tests for the `gzip` specific part - `gzip` accept headers and the fact that body is correctly compressed.

Reply:
For each of the tests, while redundant, they use the GzipEnabledConnection and tests all the same stuff but with the gzip class.